### ISSUE TITLE
opentelemetry-instrumentation-urllib log warning on None status code

### ISSUE THREAD

**Issue by jaroslawporada** [open-telemetry/opentelemetry-python-contrib#1401]
**Describe your environment** Describe any aspect of your environment relevant to the problem, including your Python version, [platform](https://docs.python.org/3/library/platform.html), version numbers of installed dependencies, information about your cloud hosting provider, etc. If you're reporting a problem with a specific version of a library in this repo, please check whether the problem has been fixed on main.

python:3.10.7

WeasyPrint==52.4
opentelemetry-instrumentation-urllib==0.33b0

'Linux-5.15.0-52-generic-x86_64-with-glibc2.28'

**Steps to reproduce**
Describe exactly how to reproduce the error. Include a code sample if applicable.

In my code following log occured `Invalid type NoneType for attribute value. Expected one of ['bool', 'str', 'bytes', 'int', 'float'] or a sequence of those types`.

It was caused by 
```
from urllib.request import Request, urlopen

response = urlopen(
   ...:     Request(
   ...:         'data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMjQiIGhlaWdodD0iMjQiIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyIgZmlsbC1ydWxlPSJldmVub2RkIiBjbGlwLXJ1bGU9ImV2ZW5vZGQiPjxwYXRoIGQ9Ik0yMSA2LjI4NWwtMTEuMTYgMTIuNzMzLTYuODQtNi4wMTggMS4zMTktMS40OSA1LjM0MSA0LjY4NiA5Ljg2NS0xMS4xOTYgMS40NzUgMS4yODV6Ii8+PC9zdmc+',
   ...:         {'User-Agent': 'WeasyPrint 52.4 (http://weasyprint.org/)',
   ...:          'Accept': '*/*',
   ...:          'Accept-Encoding': 'gzip, deflate'},
   ...:     ),
   ...:     timeout=2,
   ...:     context=None,
)
```

response.code is None. I think that in this case we should not assume that status code is not None.

**What is the expected behavior?**
 `Invalid type NoneType for attribute value. Expected one of ['bool', 'str', 'bytes', 'int', 'float'] or a sequence of those types` should not be logged in this case.

**What is the actual behavior?**
What did you see instead?

 `Invalid type NoneType for attribute value. Expected one of ['bool', 'str', 'bytes', 'int', 'float'] or a sequence of those types`  is logged.
 
**Additional context**

**Comment by shalevr** [comment]
Can you assign it to me?

**Comment by shalevr** [comment]
I'm not sure if we should add the response.code to the span in this case.
I think there is no reason to add an attribute that doesn't exist(None)

**Comment by srikanthccv** [comment]
>so it will be more complicated to change this only for the response.code.

This issue is specially about the urllib instrumentation and the fix is to not set the status code where there is None.

**Comment by srikanthccv** [comment]
Fixed in #1430
